ranger-http/src/main/java/com/flipkart/ranger/http/servicefinder/HttpNodeDataSource.java

```diff
@@ -23,59 +41,66 @@
  */
 @Slf4j
 public class HttpNodeDataSource<T, D extends HTTPResponseDataDeserializer<T>> extends HttpNodeDataStoreConnector<T> implements NodeDataSource<T, D> {
+
+    private final Service service;
+
     public HttpNodeDataSource(
             Service service,
             final HttpClientConfig config,
             ObjectMapper mapper) {
-        super(service, config, mapper);
+        super(config, mapper);
+        this.service = service;
     }
 
-
     @Override
     public Optional<List<ServiceNode<T>>> refresh(D deserializer) {
+        Preconditions.checkNotNull(config, "client config has not been set for node data");
+        Preconditions.checkNotNull(mapper, "mapper has not been set for node data");
         val httpUrl = new HttpUrl.Builder()
                 .scheme(config.isSecure()
                         ? "https"
                         : "http")
                 .host(config.getHost())
                 .port(config.getPort() == 0
-                      ? defaultPort()
-                      : config.getPort())
+                        ? defaultPort()
+                        : config.getPort())
                 .encodedPath(String.format("/ranger/nodes/v1/%s/%s", service.getNamespace(), service.getServiceName()))
                 .build();
         val request = new Request.Builder()
                 .url(httpUrl)
                 .get()
                 .build();
+
+        List<ServiceNode<T>> serviceNodes = Collections.emptyList();
         try (val response = httpClient.newCall(request).execute()) {
             if (response.isSuccessful()) {
                 try (final ResponseBody body = response.body()) {
                     if (null == body) {
                         log.warn("HTTP call to {} returned empty body", httpUrl);
-                    }
-                    else {
+                    } else {
                         final byte[] bytes = body.bytes();
-                        return Optional.of(FinderUtils.filterValidNodes(
-                                service,
-                                deserializer.deserialize(bytes),
-                                healthcheckZombieCheckThresholdTime(service)));
+                        val serviceNodesResponse = deserializer.deserialize(bytes);
+                        if(serviceNodesResponse.isSuccess()){
+                            serviceNodes = FinderUtils.filterValidNodes(
+                                    service,
+                                    serviceNodesResponse.getData(),
+                                    healthcheckZombieCheckThresholdTime(service));
+                        }else{
```

Comment:
all else in next line

Reply:
The convention seems to be space else in the code. Followed the same here too.